test/csuite/wt12015_backup_corruption/main.c

```diff
@@ -0,0 +1,442 @@
+
+        /* Die before finishing the next checkpoint. */
+        printf("Setting the failpoint...\n");
+        testutil_check(
+          session->reconfigure(session, "debug=(checkpoint_fail_before_turtle_update=true)"));
+        testutil_sentinel(NULL, EXPECT_ABORT);
+        testutil_backup_force_stop(session);
+        testutil_remove(EXPECT_ABORT);
+
+        /* We should die before we get here. */
+        testutil_die(ENOTRECOVERABLE, "The child process was supposed be dead by now!");
+    }
+
+    /* Parent. */
+
+    /* Wait for the child to die. */
+    testutil_assert(waitpid(pid, &status, 0) > 0);
+    printf("-- crash --\n");
+
+    /* Save the database directory. */
+    testutil_copy(WT_HOME_DIR, "save");
+
+    /* Do more work, create another backup, and verify it. */
+    do_work_after_failure(false);
+}
+
+/*
+ * usage --
+ *     Print usage help for the program.
+ */
+static void
+usage(void)
+{
+    fprintf(stderr, "usage: %s%s [-n] [-s SCENARIO_NUM]\n\n", progname, opts->usage);
+    fprintf(stderr, "Test-specific options:\n");
+    fprintf(stderr, "    -n    disable log\n");
+    fprintf(stderr, "    -s N  set the scenario number (use 0 for all scenarios)\n");
+    exit(EXIT_FAILURE);
+}
+
+/*
+ * main --
+ *     The entry point for the test.
+ */
+int
+main(int argc, char *argv[])
+{
+    struct sigaction sa;
+    int ch, scenario;
+    char start_cwd[PATH_MAX];
+
+    (void)testutil_set_progname(argv);
+
+    /* Automatically flush after each newline, so that we don't miss any messages if we crash. */
+    __wt_stream_set_line_buffer(stderr);
+    __wt_stream_set_line_buffer(stdout);
+
+    opts = &_opts;
+    memset(opts, 0, sizeof(*opts));
+
+    env_config = ENV_CONFIG_DEFAULT;
+    /* FIXME-WT-11669 Enable the tests by setting the following variable to 0. */
+    scenario = -1;
+
+    /* Parse the command-line arguments. */
+    testutil_parse_begin_opt(argc, argv, SHARED_PARSE_OPTIONS, opts);
+    while ((ch = __wt_getopt(progname, argc, argv, "ns:" SHARED_PARSE_OPTIONS)) != EOF)
+        switch (ch) {
+        case 'n':
+            env_config = ENV_CONFIG_NOLOG;
+            break;
+        case 's':
+            scenario = atoi(__wt_optarg);
+            break;
+        default:
+            if (testutil_parse_single_opt(opts, ch) != 0)
+                usage();
+        }
+    argc -= __wt_optind;
+    if (argc != 0)
+        usage();
+
+    testutil_parse_end_opt(opts);
+    testutil_work_dir_from_path(home, sizeof(home), opts->home);
+
+    /* Create the test directory. */
+    testutil_recreate_dir(home);
+    testutil_assert_errno(getcwd(start_cwd, sizeof(start_cwd)) != NULL);
+    testutil_assert_errno(chdir(home) == 0);
+
+    /* Configure the child death handling. */
+    memset(&sa, 0, sizeof(sa));
+    sa.sa_handler = handler_sigchld;
+    testutil_assert_errno(sigaction(SIGCHLD, &sa, NULL) == 0);
+
+    /* Run the tests. */
+    if (scenario == 0 || scenario == 1)
```

Comment:
I'd make the 1 and 2 defined macros although then the scenario arg of `-s 1` or `-s 2` isn't as obvious. I'd still suggest changing the function names so someone looking at the code sees what 1 and 2 actually test here.

Reply:
Done.